src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -678,7 +678,7 @@ protected Class impl(Class interfce) {
      * Custom reflection provider which unwraps proxies, and skips empty collections
      * and maps.
      */
-    class CustomReflectionProvider extends Sun14ReflectionProvider {
+    class CustomReflectionProvider extends SunUnsafeReflectionProvider {
```

Comment:
We were extending a old and deprecated class, the new one is somewhat faster and it's supported.